File: frontend/src/shared/models/database/flow-location-dto.ts
import { AnchorTypeEnum } from "@/shared/enums/backend/area/anchor-type-enum";
import { AreaSizingModeEnum } from "@/shared/enums/backend/area/area-sizing-mode-enum";

export class FlowLocationDto {
  id: number = 0;
  name: string = "";

  // Frame this point lives in. Null = absolute screen, which needs rebinding after an import.
  flowSearchAreaId?: number | null;
  anchor: AnchorTypeEnum = AnchorTypeEnum.TOP_LEFT;
  offsetMode: AreaSizingModeEnum = AreaSizingModeEnum.ABSOLUTE_PX;

  locationX: number = 0;
  locationY: number = 0;

  ratioX: number = 0;
  ratioY: number = 0;

  flowId: number = 0;

  // Read only, set by the backend.
  flowStepsCount: number = 0;
  flowSearchAreaName: string = "";

  constructor(data: Partial<FlowLocationDto> = {}) {
    Object.assign(this, {
      ...data,
    });
  }
}

File: backend/Core/Models/Dtos/Database/FlowLocationDto.cs
using Core.Enums;

namespace Core.Models.Dtos
{
    public class FlowLocationDto
    {
        /// <summary>0 for a new row. Negative for a new row that references another new row.</summary>
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public int? FlowSearchAreaId { get; set; }
        public AreaSizingModeEnum OffsetMode { get; set; }

        public int LocationX { get; set; }
        public int LocationY { get; set; }

        public float RatioX { get; set; }
        public float RatioY { get; set; }

        public int FlowId { get; set; }

        // Read only, projected by the query.
        public int FlowStepsCount { get; set; }
        public string FlowSearchAreaName { get; set; } = string.Empty;
    }
}
